File: leetcode/python/1557_minimum_number_of_vertices_to_reach_all_nodes.py
class Solution:
    def findSmallestSetOfVertices(self, n: int, edges: list[list[int]]) -> list[int]:
        # count incoming edges for each node
        incoming = [0] * n
        for edge in edges:
            incoming[edge[1]] += 1
        # return nodes with no incoming edges
        return [i for i in range(n) if incoming[i] == 0]

File: leetcode/rust/src/1557_minimum_number_of_vertices_to_reach_all_nodes.rs

impl Solution {
    pub fn find_smallest_set_of_vertices(n: i32, edges: Vec<Vec<i32>>) -> Vec<i32> {
        let mut indegree = vec![0; n as usize];
        for edge in edges {
            indegree[edge[1] as usize] += 1;
        }

        indegree
            .iter()
            .enumerate()
            .filter(|(_, &x)| x == 0)
            .map(|(i, _)| i as i32)
            .collect()

    }
}